### PR TITLE
8245216: [lworld] The JVM should inject the IdentityObject interface to types which need it

### PR DESCRIPTION
**Pull request by fparain**
Please review these changes for JDK-8245216.

This code makes the JVM to inject the IdentityObject interface when needed.
The details are in the CR:
https://bugs.openjdk.java.net/browse/JDK-8245216

Thank you,

Fred

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8245216](https://bugs.openjdk.java.net/browse/JDK-8245216): [lworld] The JVM should inject the IdentityObject interface to types which need it ⚠️ Title mismatch between PR and JBS.

### Reviewers
 * Harold Seigel ([hseigel](@hseigel) - Committer) ⚠️ Review applies to 00cfcb6c42c94186f3786a3b807cfe0daac67f87

### Download
`$ git fetch https://git.openjdk.java.net/valhalla pull/51/head:pull/51`
`$ git checkout pull/51`

### CODE REVIEW

**Comment by fparain** [comment]
Got some crashes during testing, investigating...

**Comment by sadayapalam** [comment]
This patch addresses the two langtools failure it should. Thanks!

However, hotspot/jtreg/runtime/valhalla/valuetypes/testSupers/TestSuperClasses.java is still referencing withdrawn classes and fails. Are you able to include a fix for that too ? 

I looked into preparing a patch, but not familiar with jcod files...

**Comment by fparain** [comment]
The PR has bee updated with the following changes:
  - fix JVMTI class file reconstruction to exclude injected interfaces
  - fix java.lang.Class and TestSupers tests
  - fix JVMCI implementation and test

**Comment by fparain** [comment]
Stress testing shown an issue with metaspace allocation/deallocation.
Withdrawing the PR until the issue has been solved.

File: src/hotspot/share/classfile/verificationType.cpp [hunk 1]
```diff
@@ -64,7 +64,8 @@ bool VerificationType::resolve_and_check_assignability(InstanceKlass* klass, Sym
     // Otherwise, we treat interfaces as java.lang.Object.
     return !from_is_array ||
       this_class == SystemDictionary::Cloneable_klass() ||
```
Comment (hseigel) [inline src/hotspot/share/classfile/verificationType.cpp:66]: Can you update the comment at line 63 to include java.lang.IdentityObject?

**Comment by fparain** [comment]
Thank you Harold. I've updated the comment in verificationType.cpp in the last commit.

Fred

**Pull request closed without merging**